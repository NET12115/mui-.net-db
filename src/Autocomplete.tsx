import React, { ChangeEvent } from 'react';

import { Field, FieldRenderProps, FieldProps } from 'react-final-form';

import TextField, { TextFieldProps as MuiTextFieldProps } from '@material-ui/core/TextField';
import {
	default as MuiAutocomplete,
	AutocompleteProps as MuiAutocompleteProps,
	RenderInputParams as MuiAutocompleteRenderInputParams,
} from '@material-ui/lab/Autocomplete';

export type AutocompleteData = {
	[key: string]: any;
};

interface AutocompleteProps extends Partial<MuiAutocompleteProps> {
	name: string;
	label: string;
	required?: boolean;
	getOptionValue: (option: any) => any;
	options: AutocompleteData[];
	fieldProps?: FieldProps<any, any>;
	textFieldProps?: Partial<MuiTextFieldProps>;
}

export const Autocomplete = (props: AutocompleteProps) => {
	const { name, fieldProps, ...rest } = props;

	return (
		<Field
			name={name}
			render={fieldRenderProps => <AutocompleteWrapper {...fieldRenderProps} {...rest} />}
			{...fieldProps}
		/>
	);
};

interface MuiRffAutocompleteProps extends FieldRenderProps<MuiTextFieldProps, HTMLElement> {
	label: string;
	required?: boolean;
	multiple?: boolean;
	textFieldProps?: Partial<MuiTextFieldProps>;
	getOptionValue: (option: any) => any;
}

const AutocompleteWrapper = (props: MuiRffAutocompleteProps) => {
	const {
		input: { name, onChange, value, ...restInput },
		meta,
		label,
		required,
		multiple,
		textFieldProps,
		getOptionValue,
		...rest
	} = props;

	const showError = ((meta.submitError && !meta.dirtySinceLastSubmit) || meta.error) && meta.touched;
	const { variant, ...restTextFieldProps } = (textFieldProps as any) || {};

	return (
		<MuiAutocomplete
			multiple={multiple}
			onChange={
				multiple
					? (_e: ChangeEvent<{}>, values: any[]): void => {
							onChange(values.map(getOptionValue));
					  }
					: (_e: ChangeEvent<{}>, option: any): void => {
							option ? onChange(getOptionValue(option)) : onChange(undefined);
					  }
			}
			renderInput={(params: MuiAutocompleteRenderInputParams) => (
				<TextField
					label={label}
					required={required}
					margin="normal"
					fullWidth
					error={showError}
					helperText={showError ? meta.error || meta.submitError : undefined}
					variant={variant}
					{...params}
					{...restInput}
					{...restTextFieldProps}
				/>
			)}
			{...rest}
		/>
	);
};
