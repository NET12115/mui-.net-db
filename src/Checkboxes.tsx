import {
	Checkbox as MuiCheckbox,
	FormControl,
	FormControlLabel,
	FormGroup,
	FormHelperText,
	FormLabel,
} from '@material-ui/core';
import { CheckboxProps } from '@material-ui/core/Checkbox';
import { FormControlProps } from '@material-ui/core/FormControl';
import React from 'react';

import { Field, FieldRenderProps } from 'react-final-form';

export interface CheckboxData {
	label: string;
	value: string;
}

export interface CheckboxesProps {
	required: boolean;
	label: string;
	name: string;
	data: CheckboxData[];
	error: string;
	fieldProps?: FieldRenderProps<CheckboxProps, HTMLInputElement>;
	formControlProps?: FormControlProps;
}

export function Checkboxes(props: CheckboxesProps) {
	const {
		required,
		label,
		data,
		name,
		error,
		fieldProps,
		formControlProps,
	} = props;

	return (
		<FormControl
			required={required}
			error={!!error}
			margin="normal"
			{...formControlProps}
		>
			<FormLabel>{label}</FormLabel>
			<FormGroup>
				{data.map((item: CheckboxData, idx: number) => (
					<FormControlLabel
						key={idx}
						label={item.label}
						value={item.value}
						control={
							<Field
								component={CheckboxWrapper}
								type="checkbox"
								name={name}
								{...fieldProps}
							/>
						}
					/>
				))}
			</FormGroup>
			{error ? <FormHelperText>{error}</FormHelperText> : <></>}
		</FormControl>
	);
}

function CheckboxWrapper(
	props: FieldRenderProps<CheckboxProps, HTMLInputElement>
) {
	const {
		input: { name, checked, onChange, ...restInput },
		meta,
		...rest
	} = props;
	return (
		<MuiCheckbox
			name={name}
			checked={checked}
			onChange={onChange}
			inputProps={restInput as any}
			{...rest}
		/>
	);
}
