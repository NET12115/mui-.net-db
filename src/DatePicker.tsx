import React from 'react';

import 'date-fns';
import DateFnsUtils from '@date-io/date-fns';

import {
	KeyboardDatePicker,
	KeyboardDatePickerProps,
	MuiPickersUtilsProvider,
} from '@material-ui/pickers';

import { Field, FieldRenderProps } from 'react-final-form';

export function DatePicker(props: Partial<KeyboardDatePickerProps>) {
	const { name } = props;

	return (
		<Field
			name={name as any}
			render={({ input, meta }) => (
				<DatePickerWrapper input={input} meta={meta} {...props} />
			)}
		/>
	);
}

function DatePickerWrapper(
	props: FieldRenderProps<KeyboardDatePickerProps, HTMLElement>
) {
	const {
		input: { name, onChange, value, ...restInput },
		meta,
		...rest
	} = props;
	const showError =
		((meta.submitError && !meta.dirtySinceLastSubmit) || meta.error) &&
		meta.touched;

	return (
		<MuiPickersUtilsProvider utils={DateFnsUtils}>
			<KeyboardDatePicker
				disableToolbar
				fullWidth={true}
				autoOk={true}
				helperText={showError ? meta.error || meta.submitError : undefined}
				error={showError}
				variant="inline"
				format="yyyy-MM-dd"
				margin="normal"
				onChange={onChange}
				value={(value as any) === '' ? null : value}
				{...rest}
				inputProps={restInput}
			/>
		</MuiPickersUtilsProvider>
	);
}
