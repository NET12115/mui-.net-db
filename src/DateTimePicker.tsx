import React from 'react';

import { DateTimePicker as MuiDateTimePicker, DateTimePickerProps as MuiDateTimePickerProps } from '@mui/lab';

import { Field, FieldProps, FieldRenderProps } from 'react-final-form';

import { ShowErrorFunc, showErrorOnChange } from './Util';
import { TextField } from '@mui/material';
import pickerProviderWrapper from './PickerProvider';

export interface DateTimePickerProps extends Partial<Omit<MuiDateTimePickerProps, 'onChange'>> {
	name: string;
	dateFunsUtils?: any;
	locale?: any;
	fieldProps?: Partial<FieldProps<any, any>>;
	showError?: ShowErrorFunc;
}

export function DateTimePicker(props: DateTimePickerProps) {
	const { name, fieldProps, ...rest } = props;

	return (
		<Field
			name={name}
			render={(fieldRenderProps) => <DateTimePickerWrapper {...fieldRenderProps} {...rest} />}
			{...fieldProps}
		/>
	);
}

interface DateTimePickerWrapperProps extends FieldRenderProps<MuiDateTimePickerProps> {
	dateFunsUtils?: any;
	locale?: any;
}

function DateTimePickerWrapper(props: DateTimePickerWrapperProps) {
	const {
		input: { name, onChange, value, ...restInput },
		meta,
		dateFunsUtils,
		locale,
		showError = showErrorOnChange,
		...rest
	} = props;

	const { error, submitError } = meta;
	const isError = showError({ meta });

	const { helperText, ...lessrest } = rest;

	return pickerProviderWrapper(
		dateFunsUtils,
		<MuiDateTimePicker
			onChange={onChange}
			value={(value as any) === '' ? null : value}
			{...lessrest}
			renderInput={(props) => (
				<TextField
					fullWidth={true}
					helperText={isError ? error || submitError : helperText}
					error={isError}
					name={name}
					{...restInput}
					{...props}
				/>
			)}
		/>,
		locale,
	);
}
