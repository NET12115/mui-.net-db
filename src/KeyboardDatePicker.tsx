import React from 'react';

import {
	KeyboardDatePicker as MuiKeyboardDatePicker,
	KeyboardDatePickerProps as MuiKeyboardDatePickerProps,
	MuiPickersUtilsProvider,
} from '@material-ui/pickers';

import { Field, FieldRenderProps } from 'react-final-form';

interface KeyboardDatePickerProps extends Partial<MuiKeyboardDatePickerProps> {
	dateFunsUtils: any;
}

export function KeyboardDatePicker(props: KeyboardDatePickerProps) {
	const { name } = props;

	return (
		<Field
			name={name as any}
			render={fieldRenderProps => (
				<KeyboardDatePickerWrapper {...fieldRenderProps} {...props} />
			)}
		/>
	);
}

interface DatePickerWrapperProps
	extends FieldRenderProps<MuiKeyboardDatePickerProps, HTMLElement> {
	dateFunsUtils: any;
}

function KeyboardDatePickerWrapper(props: DatePickerWrapperProps) {
	const {
		input: { name, onChange, value, ...restInput },
		meta,
		dateFunsUtils,
		...rest
	} = props;

	const showError =
		((meta.submitError && !meta.dirtySinceLastSubmit) || meta.error) &&
		meta.touched;

	return (
		<MuiPickersUtilsProvider utils={dateFunsUtils}>
			<MuiKeyboardDatePicker
				disableToolbar
				fullWidth={true}
				autoOk={true}
				helperText={showError ? meta.error || meta.submitError : undefined}
				error={showError}
				variant="inline"
				format="yyyy-MM-dd"
				margin="normal"
				onChange={onChange}
				value={(value as any) === '' ? null : value}
				{...rest}
				inputProps={restInput}
			/>
		</MuiPickersUtilsProvider>
	);
}
