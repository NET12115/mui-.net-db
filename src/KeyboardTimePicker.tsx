import React from 'react';

import {
	KeyboardTimePicker as MuiKeyboardTimePicker,
	KeyboardTimePickerProps as MuiKeyboardTimePickerProps,
} from '@material-ui/pickers';

import { Field, FieldProps, FieldRenderProps } from 'react-final-form';

import pickerProviderWrapper from './PickerProvider';

export interface KeyboardTimePickerProps extends Partial<MuiKeyboardTimePickerProps> {
	dateFunsUtils?: any;
	fieldProps?: FieldProps<any, any>;
}

export function KeyboardTimePicker(props: KeyboardTimePickerProps) {
	const { name, fieldProps, ...rest } = props;

	return (
		<Field
			name={name as any}
			render={fieldRenderProps => <KeyboardTimePickerWrapper {...fieldRenderProps} {...rest} />}
			{...fieldProps}
		/>
	);
}

interface KeyboardTimePickerWrapperProps extends FieldRenderProps<MuiKeyboardTimePickerProps, HTMLElement> {
	dateFunsUtils?: any;
}

function KeyboardTimePickerWrapper(props: KeyboardTimePickerWrapperProps) {
	const {
		input: { name, onChange, value, ...restInput },
		meta,
		dateFunsUtils,
		...rest
	} = props;

	const showError = ((meta.submitError && !meta.dirtySinceLastSubmit) || meta.error) && meta.touched;

	return pickerProviderWrapper(
		dateFunsUtils,
		<MuiKeyboardTimePicker
			fullWidth={true}
			autoOk={true}
			helperText={showError ? meta.error || meta.submitError : undefined}
			error={showError}
			margin="normal"
			onChange={onChange}
			name={name}
			value={(value as any) === '' ? null : value}
			{...rest}
			inputProps={restInput}
		/>
	);
}
