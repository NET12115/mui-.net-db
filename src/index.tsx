export { Checkboxes } from './Checkboxes';
export { CheckboxData } from './Checkboxes';

export { DatePicker } from './DatePicker';

export { makeValidate } from './Validation';
