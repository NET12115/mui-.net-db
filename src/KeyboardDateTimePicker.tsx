import { DateTimePicker, DateTimePickerProps } from './DateTimePicker';

/**
 * @deprecated use DateTimePicker instead
 */
export const KeyboardDateTimePicker = DateTimePicker;

/**
 * @deprecated use DateTimePickerProps instead
 */
export type KeyboardDateTimePickerProps = DateTimePickerProps;
