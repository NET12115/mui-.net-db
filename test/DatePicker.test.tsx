import { createGenerateClassName, StylesProvider } from '@material-ui/core/styles';
import { render } from '@testing-library/react';
import React from 'react';
import { Form } from 'react-final-form';

import 'date-fns';
import DateFnsUtils from '@date-io/date-fns';

import { DatePicker } from '../src';

interface ComponentProps {
	initialValues: FormData;
	validator?: any;
}

interface FormData {
	date: Date;
}

describe('DatePicker', () => {
	const defaultDateString = '2019-10-18';

	const initialValues: FormData = {
		date: new Date(defaultDateString),
	};

	function DatePickerComponent({ initialValues, validator }: ComponentProps) {
		// make a copy of the data because the state is mutated below in one of the tests for clicks
		// then the state is used again for comparison later, which causes tests to be dependent on execution
		// order and fail.
		const generateClassName = createGenerateClassName({
			disableGlobal: true,
			productionPrefix: 'test',
		});

		const onSubmit = (values: FormData) => {
			console.log(values);
		};

		const validate = async (values: FormData) => {
			if (validator) {
				return validator(values);
			}
		};

		return (
			<StylesProvider generateClassName={generateClassName}>
				<Form
					onSubmit={onSubmit}
					initialValues={initialValues}
					validate={validate}
					render={({ handleSubmit }) => (
						<form onSubmit={handleSubmit} noValidate>
							<DatePicker label="Test" name="date" required={true} dateFunsUtils={DateFnsUtils} />
						</form>
					)}
				/>
			</StylesProvider>
		);
	}

	it('renders without errors', () => {
		const rendered = render(<DatePickerComponent initialValues={initialValues} />);
		expect(rendered).toMatchSnapshot();
	});

	it('renders the value with default data', async () => {
		const rendered = render(<DatePickerComponent initialValues={initialValues} />);
		const date = (await rendered.findByDisplayValue(defaultDateString)) as HTMLInputElement;
		expect(date.value).toBe(defaultDateString);
	});

	it('has the Test label', () => {
		const rendered = render(<DatePickerComponent initialValues={initialValues} />);
		const elem = rendered.getByText('Test') as HTMLLegendElement;
		expect(elem.tagName).toBe('LABEL');
	});

	it('has the required *', () => {
		const rendered = render(<DatePickerComponent initialValues={initialValues} />);
		const elem = rendered.getByText('*') as HTMLSpanElement;
		expect(elem.tagName).toBe('SPAN');
		expect(elem.innerHTML).toBe('â€‰*');
	});
});
