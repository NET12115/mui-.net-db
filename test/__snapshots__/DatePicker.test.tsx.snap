// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker renders without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        novalidate=""
      >
        <div
          class="MuiFormControl-root-2 MuiTextField-root-1 MuiFormControl-marginNormal-3 MuiFormControl-fullWidth-5"
        >
          <label
            class="MuiFormLabel-root-18 MuiInputLabel-root-6 MuiInputLabel-formControl-12 MuiInputLabel-animated-15 MuiInputLabel-shrink-14 MuiFormLabel-filled-22 MuiFormLabel-required-23 MuiInputLabel-required-10"
            data-shrink="true"
          >
            Test
            <span
              class="MuiFormLabel-asterisk-24 MuiInputLabel-asterisk-11"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root-37 MuiInput-root-25 MuiInput-underline-29 MuiInputBase-fullWidth-46 MuiInput-fullWidth-32 MuiInputBase-formControl-38 MuiInput-formControl-26 MuiInputBase-adornedEnd-42"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input-47 MuiInput-input-33 MuiInputBase-inputAdornedEnd-55"
              name="date"
              required=""
              value="2019-10-18"
            />
            <div
              class="MuiInputAdornment-root-57 MuiInputAdornment-positionEnd-60"
            >
              <button
                class="MuiButtonBase-root-73 MuiIconButton-root-64"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label-72"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root-76"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root-87"
                />
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      novalidate=""
    >
      <div
        class="MuiFormControl-root-2 MuiTextField-root-1 MuiFormControl-marginNormal-3 MuiFormControl-fullWidth-5"
      >
        <label
          class="MuiFormLabel-root-18 MuiInputLabel-root-6 MuiInputLabel-formControl-12 MuiInputLabel-animated-15 MuiInputLabel-shrink-14 MuiFormLabel-filled-22 MuiFormLabel-required-23 MuiInputLabel-required-10"
          data-shrink="true"
        >
          Test
          <span
            class="MuiFormLabel-asterisk-24 MuiInputLabel-asterisk-11"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root-37 MuiInput-root-25 MuiInput-underline-29 MuiInputBase-fullWidth-46 MuiInput-fullWidth-32 MuiInputBase-formControl-38 MuiInput-formControl-26 MuiInputBase-adornedEnd-42"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input-47 MuiInput-input-33 MuiInputBase-inputAdornedEnd-55"
            name="date"
            required=""
            value="2019-10-18"
          />
          <div
            class="MuiInputAdornment-root-57 MuiInputAdornment-positionEnd-60"
          >
            <button
              class="MuiButtonBase-root-73 MuiIconButton-root-64"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label-72"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-76"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                  />
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root-87"
              />
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`DatePicker requires a date value 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        novalidate=""
      >
        <div
          class="MuiFormControl-root-2 MuiTextField-root-1 MuiFormControl-marginNormal-3 MuiFormControl-fullWidth-5"
        >
          <label
            class="MuiFormLabel-root-18 MuiInputLabel-root-6 MuiInputLabel-formControl-12 MuiInputLabel-animated-15 MuiFormLabel-error-21 MuiInputLabel-error-9 MuiFormLabel-required-23 MuiInputLabel-required-10"
            data-shrink="false"
          >
            Test
            <span
              class="MuiFormLabel-asterisk-24 MuiInputLabel-asterisk-11 MuiFormLabel-error-21 MuiInputLabel-error-9"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root-37 MuiInput-root-25 MuiInput-underline-29 MuiInputBase-error-43 MuiInput-error-30 MuiInputBase-fullWidth-46 MuiInput-fullWidth-32 MuiInputBase-formControl-38 MuiInput-formControl-26 MuiInputBase-adornedEnd-42"
          >
            <input
              aria-invalid="true"
              class="MuiInputBase-input-47 MuiInput-input-33 MuiInputBase-inputAdornedEnd-55"
              name="date"
              required=""
              value=""
            />
            <div
              class="MuiInputAdornment-root-57 MuiInputAdornment-positionEnd-60"
            >
              <button
                class="MuiButtonBase-root-73 MuiIconButton-root-64"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label-72"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root-76"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root-87"
                />
              </button>
            </div>
          </div>
          <p
            class="MuiFormHelperText-root-97 MuiFormHelperText-error-98 MuiFormHelperText-required-104"
          >
            something for testing
          </p>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      novalidate=""
    >
      <div
        class="MuiFormControl-root-2 MuiTextField-root-1 MuiFormControl-marginNormal-3 MuiFormControl-fullWidth-5"
      >
        <label
          class="MuiFormLabel-root-18 MuiInputLabel-root-6 MuiInputLabel-formControl-12 MuiInputLabel-animated-15 MuiFormLabel-error-21 MuiInputLabel-error-9 MuiFormLabel-required-23 MuiInputLabel-required-10"
          data-shrink="false"
        >
          Test
          <span
            class="MuiFormLabel-asterisk-24 MuiInputLabel-asterisk-11 MuiFormLabel-error-21 MuiInputLabel-error-9"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root-37 MuiInput-root-25 MuiInput-underline-29 MuiInputBase-error-43 MuiInput-error-30 MuiInputBase-fullWidth-46 MuiInput-fullWidth-32 MuiInputBase-formControl-38 MuiInput-formControl-26 MuiInputBase-adornedEnd-42"
        >
          <input
            aria-invalid="true"
            class="MuiInputBase-input-47 MuiInput-input-33 MuiInputBase-inputAdornedEnd-55"
            name="date"
            required=""
            value=""
          />
          <div
            class="MuiInputAdornment-root-57 MuiInputAdornment-positionEnd-60"
          >
            <button
              class="MuiButtonBase-root-73 MuiIconButton-root-64"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label-72"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-76"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                  />
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root-87"
              />
            </button>
          </div>
        </div>
        <p
          class="MuiFormHelperText-root-97 MuiFormHelperText-error-98 MuiFormHelperText-required-104"
        >
          something for testing
        </p>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
